src/geometry_aux.cpp

```diff
@@ -167,26 +167,36 @@ get_temperatures(std::vector<std::vector<double>>& nuc_temps,
       int i_material = cell->material_[j];
       if (i_material == MATERIAL_VOID) continue;
 
-      // Get temperature of cell (rounding to nearest integer)
-      double sqrtkT = cell->sqrtkT_.size() == 1 ?
-        cell->sqrtkT_[j] : cell->sqrtkT_[0];
-      double temperature = sqrtkT*sqrtkT / K_BOLTZMANN;
+      // Get temperature(s) of cell (rounding to nearest integer)
+      std::vector<double> cell_temps;
+      if (cell->sqrtkT_.size() == 1) {
+        double sqrtkT = cell->sqrtkT_[0];
+        cell_temps.push_back(sqrtkT*sqrtkT / K_BOLTZMANN);
+      } else if (cell->sqrtkT_.size() == cell->material_.size()) {
+        double sqrtkT = cell->sqrtkT_[j];
+        cell_temps.push_back(sqrtkT*sqrtkT / K_BOLTZMANN);
+      } else {
+        for (double sqrtkT : cell->sqrtkT_)
+          cell_temps.push_back(sqrtkT*sqrtkT / K_BOLTZMANN);
+      }
```

Comment:
Seems like determining `cell_temps` ought to be done as a separate loop above the loop over `cell->material_.size()` (line 165). Effectively in each case, we're iteratating over `cell->sqrtkT_`.

Reply:
So populate the `cell_temps` array before the `cell->material_.size()` loop?  I think that way we could end up with loading more temperature data than needed.  If every cell instance has a different material and different temperature, then each material only needs to load one temperature.

Reply:
Ah yes, you're right. I felt like I might be missing something!